arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -787,25 +789,29 @@ Result RocksDBCollection::truncateWithRangeDelete(transaction::Methods& trx) {
   if (!s.ok()) {
     return rocksutils::convertStatus(s);
   }
+  // we need crash server if any failure in order to trigger recovery,
+  // as rocksdb truncate already committed here
+  [&]() noexcept {
+    // post commit sequence
+    rocksdb::SequenceNumber seq = db->GetLatestSequenceNumber() - 1;
+
+    uint64_t numDocs = _meta.numberDocuments();
+    _meta.adjustNumberDocuments(seq,
```

Comment:
`RocksDBMetadata::adjustNumberDocuments` performs a `try_emplace` into a container and thus can throw. It is not `noexcept` but called from a lambda marked `noexcept`. Is this intentional?

Reply:
Yep, I made comment which explains why.
If exception happens we have two options:
1. crash => recovery
2. make rollback for truncate on already truncated indexes and collection

2 iss impossible I think, so decided to use 1

Reply:
@jsteemann  as I'm going to merge this. Have you any concerns/suggestions with this PR?